src/jit/liveness.cpp

```diff
@@ -1687,204 +1687,208 @@ void Compiler::fgComputeLifeCall(VARSET_TP& life, GenTreeCall* call)
-                               VarSetOps::ToString(this, VarSetOps::Union(this, life,
-                                                                          VarSetOps::MakeSingleton(this, varIndex))),
-                               VarSetOps::ToString(this, life));
-                    }
+    if (verbose && 0)
+    {
+        printf("Ref V%02u,T%02u] at ", node->gtLclNum, varIndex);
+        printTreeID(node);
+        printf(" life %s -> %s\n", VarSetOps::ToString(this, life),
+               VarSetOps::ToString(this, VarSetOps::AddElem(this, life, varIndex)));
+    }
 #endif // DEBUG
-                }
-            }
-            else
-            {
-                /* Dead assignment to the variable */
-                lclVarNode->gtFlags |= GTF_VAR_DEATH;
 
-                if (!opts.MinOpts())
-                {
-                    // keepAliveVars always stay alive
-                    noway_assert(!VarSetOps::IsMember(this, keepAliveVars, varIndex));
+    // The variable is being used, and it is not currently live.
+    // So the variable is just coming to life
+    node->gtFlags |= GTF_VAR_DEATH;
+    VarSetOps::AddElemD(this, life, varIndex);
 
-                    /* This is a dead store unless the variable is marked
-                       GTF_VAR_USEASG and we are in an interior statement
-                       that will be used (e.g. while (i++) or a GT_COMMA) */
+    // Record interference with other live variables
+    fgMarkIntf(life, varIndex);
+}
 
-                    // Do not consider this store dead if the target local variable represents
-                    // a promoted struct field of an address exposed local or if the address
-                    // of the variable has been exposed. Improved alias analysis could allow
-                    // stores to these sorts of variables to be removed at the cost of compile
-                    // time.
-                    return !varDsc->lvAddrExposed &&
-                           !(varDsc->lvIsStructField && lvaTable[varDsc->lvParentLcl].lvAddrExposed);
-                }
-            }
+bool Compiler::fgComputeLifeTrackedLocalDef(VARSET_TP&           life,
+                                            VARSET_VALARG_TP     keepAliveVars,
+                                            LclVarDsc&           varDsc,
+                                            GenTreeLclVarCommon* node)
+{
+    assert(node != nullptr);
+    assert((node->gtFlags & GTF_VAR_DEF) != 0);
+    assert(varDsc.lvTracked);
 
-            return false;
-        }
-        else // it is a use
+    const unsigned varIndex = varDsc.lvVarIndex;
+    if (VarSetOps::IsMember(this, life, varIndex))
+    {
+        /* The variable is live */
+
+        if ((node->gtFlags & GTF_VAR_USEASG) == 0)
         {
-            // Is the variable already known to be alive?
-            if (VarSetOps::IsMember(this, life, varIndex))
+            /* Mark variable as dead from here to its closest use */
```

Comment:
Please make this a C++ style comment, while you're here.
Also, I would rephrase this as simply "Remove variable from the live set." Talking about its closest use almost makes one think we're talking about its downstream use.